spec/models/factory_log_spec.rb

```diff
@@ -1,97 +1,29 @@
-
-    let!(:a_sample) { factoryAs.first }
-    let!(:a_with_traits_sample) { factoryAs_with_traits.first }
-    let!(:b_sample) { factoryBs.first }
-
-    let!(:a_ranking_data) {
-                            {
-                              name: (a_sample.name + a_sample.traits),
-                              total_time: 2.0 * 10,
-                              count: 10,
-                              average_time: 2.0
-                            }
-                          }
-    let!(:a_with_traits_ranking_data) {
-                                        {
-                                          name: (a_with_traits_sample.name + a_with_traits_sample.traits),
-                                          total_time: 2.0,
-                                          count: 1,
-                                          average_time: 2.0
-                                        }
-                                      }
-    let!(:b_ranking_data) {
-                            {
-                              name: (b_sample.name + b_sample.traits),
-                              total_time: 10.0 * 3,
-                              count: 3,
-                              average_time: 10.0
-                            }
-                          }
-
-    it 'ranking about time they took (total)' do
-      expect(FactoryLog.time_ranking).to eq [
-                                              b_ranking_data,
-                                              a_ranking_data,
-                                              a_with_traits_ranking_data
-                                            ]
+  describe "time ranking" do
+    let!(:inspected_factory1) { { name: "factory1", traits:[], assos: [] } }
+    let!(:inspected_factory2) { { name: "factory1", traits:["trait_a"], assos: [] } }
+    let!(:factory1) { Factory.create_unique_factory(inspected_factory1) }
+    let!(:factory2) { Factory.create_unique_factory(inspected_factory2) }
+    it do
+      FactoryLog.logging(inspected_factory1, 1.0)
+      FactoryLog.logging(inspected_factory1, 1.1)
+      FactoryLog.logging(inspected_factory1, 1.0)
+      FactoryLog.logging(inspected_factory2, 2.1)
+      expect(FactoryLog.ranking).to eq [
+                                          {
+                                            factory: factory1,
+                                            total_time: 1.0 + 1.1 + 1.0,
+                                            count: 3,
+                                            average_time: (1.0 + 1.1 + 1.0) / 3
+                                          },
+                                          {
+                                            factory: factory2,
+                                            total_time: 2.1,
+                                            count: 1,
+                                            average_time: 2.1
+                                          }
+                                       ]
```

Comment:
Style/IndentArray: Indent the right bracket the same as the start of the line where the left bracket is.